src/type-factory/object.type-factory.ts

```diff
@@ -46,7 +46,7 @@ export function objectTypeFactory(target: Function, isInput: boolean = false): g
                 SchemaFactoryErrorType.NO_FIELD);
           }
 
-          const fields = fieldMetadataList.reduce((map, fieldMetadata) => {
+          const fields = () => fieldMetadataList.reduce((map, fieldMetadata) => {
```

Comment:
🤔 clever 👏 

I will leave this SO answer here for future reference on why this works https://stackoverflow.com/a/39290345/429521